aptos-move/move-examples/nft_auction_house/sources/auction_house_own_listing.move

```diff
@@ -0,0 +1,531 @@
+        let sec = timestamp::now_seconds();
+        assert!(sec <= start_sec, error::invalid_argument(EINVALID_START_TIME));
+        assert!(start_sec < expiration_sec, error::invalid_argument(EINVALID_EXPIRATION_TIME));
+        let listing = listing::create_list<CoinType>(
+            owner,
+            token_id,
+            amount,
+            min_price,
+            false,
+            start_sec,
+            expiration_sec,
+            withdraw_expiration_sec,
+        );
+
+        Auction<CoinType>{
+            listing,
+            bids: vector::empty(),
+            highest_price: 0,
+            highest_bid: guid::create_id(@auction_house, 0),
+            min_incremental_price,
+        }
+    }
+
+    public entry fun create_auction<CoinType>(
+        owner: &signer,
+        creator: address,
+        collection_name: String,
+        token_name: String,
+        property_version: u64,
+        amount: u64,
+        min_price: u64,
+        start_sec: u64, // specify when the auction starts
+        expiration_sec: u64, // specify when the auction ends
+        min_incremental_price: u64,
+    ) acquires Auctions {
+        let token_id = token::create_token_id_raw(creator, collection_name, token_name, property_version);
+        create_auction_with_token_id<CoinType>(
+            owner,
+            token_id,
+            amount,
+            min_price,
+            start_sec,
+            expiration_sec,
+            min_incremental_price,
+        );
+    }
+
+    public fun create_auction_with_token_id<CoinType>(
+        owner: &signer,
+        token_id: TokenId,
+        amount: u64,
+        min_price: u64,
+        start_sec: u64, // specify when the auction starts
+        expiration_sec: u64, // specify when the auction ends
+        min_incremental_price: u64,
+    ): u64 acquires Auctions {
+        let auction = generate_auction_data<CoinType>(
+            owner,
+            token_id,
+            amount,
+            min_price,
+            start_sec,
+            expiration_sec,
+            expiration_sec + AUCTION_DELAY_TIME_SEC, // allow time to withdraw
+            min_incremental_price,
+        );
+
+        // initialized coin store when listing
+        if (!coin::is_account_registered<CoinType>(signer::address_of(owner))) {
+            coin::register<CoinType>(owner);
+        };
+
+        let auctions = borrow_global_mut<Auctions<CoinType>>(@auction_house);
+        event::emit_event<ListingEvent>(
+            &mut auctions.listing_event,
+            listing::create_listing_event(
+                listing::get_listing_id(&auction.listing),
+                token_id,
+                amount,
+                min_price,
+                false,
+                start_sec,
+                expiration_sec,
+                @auction_house,
+            ),
+        );
+
+        let next_id = auctions.cur_auction_id + 1;
+        *(&mut auctions.cur_auction_id) = next_id;
```

Comment:
What happens if someone creates multiple auctions at the same time? Wouldn't cur_auction_id be wrong as it only tracks the latest created auction?

Reply:
This is for generating auction_id by tracking the existing largest auction_id.